kong/client.go

```diff
@@ -280,18 +306,20 @@ func (c *Client) Status(ctx context.Context) (*Status, error) {
 	return &s, nil
 }
 
-// Root returns the response of GET request on root of
-// Admin API (GET /).
+// Root returns the response of GET request on root of Admin API (GET / or /kong with a workspace).
 func (c *Client) Root(ctx context.Context) (map[string]interface{}, error) {
-	req, err := c.NewRequest("GET", "/", nil, nil)
+	endpoint := "/"
+	if len(c.Workspace()) > 0 {
+		endpoint = "/kong"
+	}
+	req, err := c.NewRequest("GET", endpoint, nil, nil)
```

Comment:
```suggestion
	ws := c.Workspace()
	if len(ws) > 0 {
		endpoint = "/kong"
	}
	req, err := c.NewRequestRaw("GET", c.baseURL(ws), endpoint, nil, nil)
```

Like I said in a previous comment, we need to make sure that this function makes exactly one synchronized read of the workspace name. This suggestion is intended to achieve exactly that.